File: 1482-how-many-numbers-are-smaller-than-the-current-number/1482-how-many-numbers-are-smaller-than-the-current-number.go
func smallerNumbersThanCurrent(nums []int) []int {
    arr := make([]int, 101)
    res := make([]int, len(nums))
    for _, v := range nums {
        arr[v]++
    }
    for i := 1; i < 101; i++ {
        arr[i] = arr[i] + arr[i - 1]
    }
    for i := 0; i < len(nums); i++ {
        if nums[i] == 0 {
            res[i] = 0
        } else {
            res[i] = arr[nums[i] - 1]
        }
    }
    return res
}

File: 1482-how-many-numbers-are-smaller-than-the-current-number/1482-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> temp(nums.begin(), nums.end());
        sort(temp.begin(), temp.end());
        unordered_map<int, int> mp;
        for(int i=0;i<nums.size();i++){
            if(mp.find(temp[i]) == mp.end()){
                mp[temp[i]] = i;
            }
        }
        vector<int> ans;
        for(int i=0;i<nums.size();i++){
            ans.push_back(mp[nums[i]]);
        }
        return ans;
    }
};